### DISCUSSION TITLE
Add user properties

### DISCUSSION THREAD

**Discussion by davidbrochart** [frankie567/httpx-oauth#257] [Q&A]
I would like to get more information from e.g. a user's GitHub account (avatar URL, username...). What's the recommended way to do this? Since I have the `id`, should I get it from `https://api.github.com/user/:id`, or is there a way to have the user's model already populated with it?

**Comment by frankie567** [comment]
The goal of HTTPX-OAuth is to give you a valid access token for the provider API.

Once you have it, you can call the REST API separately (e.g. in a worker process or in the [`on_after_register` callback of FastAPI Users](https://frankie567.github.io/fastapi-users/configuration/routers/register/#after-register)) with this token to retrieve the information you need.
